docs/tor/tor.md

```diff
@@ -0,0 +1,143 @@
+
+To connect to onion addresses you need th enable the tor proxy.
+To do so you need to have tor currently running, this can be checked by using the command `sudo systemctl status tor`.
+This should give you an output similar to:
+
+```bash
+$ sudo systemctl status tor
+● tor.service - Anonymizing overlay network for TCP (multi-instance-master)
+     Loaded: loaded (/lib/systemd/system/tor.service; enabled; vendor preset: enabled)
+     Active: active (exited) since Wed 2021-07-28 13:06:42 CDT; 48min ago
+   Main PID: 804 (code=exited, status=0/SUCCESS)
+      Tasks: 0 (limit: 18696)
+     Memory: 0B
+     CGroup: /system.slice/tor.service
+```
+
+If the output says `Active: active`, then it is running and good to go.
+
+## Enabling the Tor proxy
+
+Enabling the tor proxy allows you to create outgoing connections over tor.
+This is needed if you want to sync the blockchain over tor, or to accept DLCs over tor.
+
+To enable the tor proxy you simply need to set a couple config options after you have tor running.
+
+You need to enable the proxy and set the host and port configuration options.
+If you are using the default settings you should only need to set `bitcoin-s.proxy.enabled = true`.
+
+```$xslt
+bitcoin-s {
+    proxy {
+        # You can configure SOCKS5 proxy to use Tor for outgoing connections
+        enabled = true
+        host = "127.0.0.1"
+        port = 9050
+    }
+}
+```
+
+## Creating our own Tor hidden service
+
+Enabling the tor hidden services allows you to create inbound connections over tor.
+This is needed if you want to create DLCs over tor.
+
+To enable the tor hidden services you need to set a couple config options after you have tor running in your bitcoin-s config, as well as
+have tor configured for it.
+
+### Configuring Tor
+
+You may need to set up the Tor Control Port. On Linux distributions there may be
+some or all of the following settings in `/etc/tor/torrc`, generally commented
+out by default (if not, add them):
+
+```
+ControlPort 9051
+CookieAuthentication 1
+CookieAuthFileGroupReadable 1
+```
+
+Add or uncomment those, save, and restart Tor (usually `systemctl restart tor`
+or `sudo systemctl restart tor` on most systemd-based systems, including recent
+Debian and Ubuntu, or just restart the computer).
+
+On some systems (such as Arch Linux), you may also need to add the following
+line:
+
+```
+DataDirectoryGroupReadable 1
+```
+
+You will also need permissions for the auth cookie file, this can be done doing
+
+```bash
+sudo usermod -a -G debian-tor $USER
+```
+
+After changing these settings, you will need to restart your computer.
+
+### Configuring Bitcoin-S
+
+You need to enable tor and set the control option, `127.0.0.1:9051` is the default.
+If you are using the default settings you should only need to set `bitcoin-s.enabled.enabled = true`.
+
+```$xslt
+bitcoin-s {
+    tor {
+        # You can enable Tor for incoming connections
+        enabled = true
+        control = "127.0.0.1:9051"
```

Comment:
nit: The port is separate above but embedded here. Not sure what is best, but probably should try to be consistent.